pysnn/neuron.py

```diff
@@ -34,38 +34,26 @@ def convert_input(self, x):
         return x.type(self.trace.dtype)
 
 
-class InputTraceExponential(Input):
-    def __init__(self, cells_shape, dt, alpha_t, tau_t):
-        super(InputTraceExponential, self).__init__(cells_shape, dt)
+class Input(BaseInput):
+    def __init__(self, cells_shape, dt, alpha_t, tau_t, update_type="linear"):
+        super(Input, self).__init__(cells_shape, dt)
         self.register_buffer("alpha_t", torch.tensor(alpha_t, dtype=torch.float))
         self.register_buffer("tau_t", torch.tensor(tau_t, dtype=torch.float))
 
-        self.init_neuron()
-
-    def update_trace(self, x):
-        x = self.convert_input(x)
-        self.trace = sf._exponential_trace_update(
-            self.trace, x, self.alpha_t, self.tau_t, self.dt
-        )
-
-    def forward(self, x):
-        self.update_trace(x)
-        return x, self.trace
-
-
-class InputTraceLinear(Input):
-    def __init__(self, cells_shape, dt, alpha_t, trace_decay):
-        super(InputTraceLinear, self).__init__(cells_shape, dt)
-        self.register_buffer("alpha_t", torch.tensor(alpha_t, dtype=torch.float))
-        self.register_buffer("trace_decay", torch.tensor(trace_decay, dtype=torch.float))
+        # TODO: here or in separate function? Or overwrite init_neuron?
```

Comment:
I prefer to put it over here, the voltage updates might differ per neuron. Also, the base class is better suited to be a container for operational functionalities while remaining as agnostic to mathematical choices as possible.